src/concepts/glean.md

```diff
@@ -0,0 +1,72 @@
+# Glean - mobile product analytics
+
+For Mozilla, getting reliable data from our products is critical to inform our decision making. Glean is our new product analytics & Telemetry solution that provides that data for our mobile products.
```

Comment:
nit: is telemetry supposed to start with a lowercase letter here?

Reply:
Also, with desktop on the roadmap, do we want to use the word "mobile" here?  (Or maybe that's too soon...)

Reply:
Currently Glean provides mobile product analytics only, so i think for now it's good to call that out.
We can say somewhere that Desktop is considered/planned, but we couldn't even put a firm date on that yet.